### PR TITLE
Fix: boolean and decimal key properties

### PR DESCRIPTION
**Pull request by ralfhandl**
- XSLT: boolean and decimal key properties
- JS: decimal key properties

### CODE REVIEW

File: tools/V4-CSDL-to-OpenAPI.xsl [hunk 1]
```diff
@@ -3862,8 +3862,9 @@
         <xsl:text>time'</xsl:text>
       </xsl:when>
       <xsl:when test="$type='Edm.Int64' or $type='Edm.Int32' or $type='Edm.Int16' or $type='Edm.SByte' or $type='Edm.Byte' 
-                   or $type='Edm.Double' or $type='Edm.Single' or $type='Edm.Date' or $type='Edm.DateTimeOffset' or $type='Edm.Guid' or $type='Edm.TimeOfDay'" />
-      <!-- TODO: handle other Edm types, enumeration types, and type definitions -->
+                   or $type='Edm.Decimal' or $type='Edm.Double' or $type='Edm.Single' or $type='Edm.Boolean' 
+                   or $type='Edm.Date' or $type='Edm.DateTimeOffset' or $type='Edm.Guid' or $type='Edm.TimeOfDay'" />
+      <!-- TODO: handle enumeration types and type definitions -->
```
Comment (HeikoTheissen) [inline tools/V4-CSDL-to-OpenAPI.xsl:3867]: What about `Edm.Duration`? In OData 4.0, this requires the prefix `duration`, and this prefix is also accepted by OData 4.01. But this transformation omits the prefix and therefore supports only 4.01.

**Comment by ralfhandl** [comment]
> Are the files tools/tests/annotations.swagger.json and tools/tests/annotations.openapi3.json generated from tools/tests/annotations.xml?

Yes

File: tools/V4-CSDL-to-OpenAPI.xsl (hunk 1 above)
Comment (ralfhandl) [reply tools/V4-CSDL-to-OpenAPI.xsl:3867]: Which flavors does RAP accept?

Comment (HeikoTheissen) [inline tools/V4-CSDL-to-OpenAPI.xsl:3867]:
```suggestion
      <xsl:when test="$odata-version='4.0' and $type='Edm.Duration'">
        <xsl:text>duration'</xsl:text>
      </xsl:when>
      <!-- TODO: handle enumeration types and type definitions -->
```

Comment (HeikoTheissen) [reply tools/V4-CSDL-to-OpenAPI.xsl:3867]: Plus a corresponding change for the closing quote.

Comment (ralfhandl) [reply tools/V4-CSDL-to-OpenAPI.xsl:3870]: Currently only with `duration` prefix

**Review by HeikoTheissen — APPROVED** [review]

**Pull request merged**